### PR TITLE
test: type check lib.deno_core.d.ts

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

**Review by ry — COMMENTED** [review]
I feel like this can be done without spawning a subprocess...

File: core/lib.deno_core.d.ts [hunk 1]
```diff
@@ -104,7 +104,7 @@ declare namespace Deno {
     ): void;
 
     /** Check if there's a scheduled "next tick". */
-    function hasNextTickScheduled(): boolean;
+    function hasNextTickScheduled(): bool;
```
[the remark this answers is not in the document]
Comment (bartlomieju) [reply core/lib.deno_core.d.ts:107]: Yes

**Comment by bartlomieju** [comment]
> I feel like this can be done without spawning a subprocess...

This is the simplest way to type check `lib.deno_core.d.ts`; other testing facilities (like in `cli/tsc.rs`) use mocked file loaders and don't provide an easy way to test that file.

File: cli/tests/integration/mod.rs [hunk 2]
```diff
@@ -1087,6 +1087,31 @@ fn typecheck_declarations_unstable() {
...
 
+#[test]
+fn typecheck_core() {
+  let deno_dir = TempDir::new().expect("tempdir fail");
+  let test_file = deno_dir.path().join("test_deno_core_types.ts");
+  std::fs::write(
+    &test_file,
+    format!(
+      "import \"{}\";",
+      util::root_path()
```
Comment (dsherret) [inline cli/tests/integration/mod.rs:1098]: It's probably doing something like `D:\something\code/lib.deno_core.d.ts` on windows.

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**